src/mpid/ch4/netmod/ofi/subconfigure.m4

```diff
@@ -211,6 +210,7 @@ AM_COND_IF([BUILD_CH4_NETMOD_OFI],[
         if test "x$ofi_direct_provider" != "x" ; then
             PAC_APPEND_FLAG([-I${master_top_builddir}/src/mpid/ch4/netmod/ofi/libfabric/prov/${ofi_direct_provider}/include], [CPPFLAGS])
             PAC_APPEND_FLAG([-I${use_top_srcdir}/src/mpid/ch4/netmod/ofi/libfabric/prov/${ofi_direct_provider}/include], [CPPFLAGS])
+            PAC_APPEND_FLAG([-DFABRIC_DIRECT],[CPPFLAGS])
```

Comment:
I think I understand now. Despite configuring libfabric for a direct provider build, you can also need to set this flag in MPICH in order to get the configuration in `fabric.h`.